### PR TITLE
[Console-Iot] Update project detail section to include token and messaging URL

### PR DESCRIPTION
**Pull request by drcoolsanjeev**
### Type of change

- Refactoring

### Description

modify project detail to include token and endpoint url in device management section

Design Url- https://marvelapp.com/5gbji22/screen/69214939

image

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [x] Resolve all warnings
- [x] Update license
- [ ] Write test cases for null checks in Unit Tests
- [ ] Reference relevant issue(s) and close them after merging

### CODE REVIEW

File: console/console-init/ui/src/modules/project-detail/components/AccessCredentials/AccessCredentials.tsx [hunk 1]
```diff
@@ -297,7 +302,10 @@ const AccessCredentials: React.FunctionComponent<IAccessCredentialsProps> = ({
       </CardHeader>
       {isHidden ? (
         <CardBody>
-          <JsonEditor readOnly={true} value={JSON.stringify(data)} />
+          <JsonEditor
+            readOnly={true}
+            value={JSON.stringify(data, undefined, 2)}
```
Comment (anukritijha) [inline console/console-init/ui/src/modules/project-detail/components/AccessCredentials/AccessCredentials.tsx:307]: Please give a descriptive name for data.

File: console/console-init/ui/src/modules/project-detail/components/DeviceRegistrationManagement/DeviceRegistrationManagement.tsx [hunk 2]
```diff
@@ -152,16 +156,16 @@ const DeviceRegistationManagement: React.FunctionComponent<IDeviceRegistationMan
...
+            </Title>
           </SplitItem>
           <SplitItem isFilled />
           <SplitItem>
-            <Switch
+            <SwitchWithToggle
               id="device-registration-management-switch"
-              label="View in Json"
-              labelOff="View in Json"
+              label="View JSON fromat"
```
Comment (anukritijha) [inline console/console-init/ui/src/modules/project-detail/components/DeviceRegistrationManagement/DeviceRegistrationManagement.tsx:167]: Typo in *format

File: console/console-init/ui/src/utils/utils.ts [hunk 3]
```diff
@@ -149,12 +149,10 @@ const hasOwnProperty = (obj: Object, property: string) => {
...
 };
 
-const getCombinedString = (a: string, b?: string) => {
-  let s: string = "";
-  s += a;
-  if (b !== undefined) {
-    s += ", ";
-    s += b;
+const getCombinedString = (list: Array<string>) => {
+  let s: string = list[0];
```
Comment (anukritijha) [inline console/console-init/ui/src/utils/utils.ts:153]: Please avoid using names like s,i, list etc

File: console/console-init/ui/src/utils/utils.ts [hunk 4]
```diff
@@ -149,14 +149,12 @@ const hasOwnProperty = (obj: Object, property: string) => {
   }
 };
 
-const getCombinedString = (a: string, b?: string) => {
-  let s: string = "";
-  s += a;
-  if (b !== undefined) {
-    s += ", ";
-    s += b;
+const getCombinedString = (list: Array<string>) => {
```
Comment (anukritijha) [inline console/console-init/ui/src/utils/utils.ts:152]: You can simply use the join() method to do this.

File: console/console-init/ui/src/utils/utils.ts [hunk 5]
```diff
@@ -181,53 +179,53 @@ const getLabelForTypeOfObject = (value: any) => {
 
 const getJsonForMetadata = (object: any, type?: string) => {
```
Comment (anukritijha) [inline console/console-init/ui/src/utils/utils.ts:180]: Can you please add some comments for these methods?

File: console/console-init/ui/yarn.lock [hunk 6]
```diff
@@ -2381,6 +2381,11 @@
   resolved "https://registry.yarnpkg.com/@types/json5/-/json5-0.0.29.tgz#ee28707ae94e11d2b827bcbe5270bcea7f3e71ee"
   integrity sha1-7ihweulOEdK4J7y+UnC86n8+ce4=
 
+"@types/lodash@^4.14.152":
```
Comment (anukritijha) [inline console/console-init/ui/yarn.lock:2384]: As discussed, please remove the lodash reference.

File: console/console-init/ui/src/utils/utils.ts [hunk 7]
```diff
@@ -211,21 +231,21 @@ const getJsonForMetadata = (object: any, type?: string) => {
   return metadataArray;
 };
 
-const getJsonForObject = (object: any) => {
+const convertObjectIntoJson = (object: any) => {
```
Comment (anukritijha) [inline console/console-init/ui/src/utils/utils.ts:234]: Please add comments for this method as well.

File: console/console-init/ui/src/modules/project-detail/components/GeneralInfo/GeneralInfo.tsx [hunk 8]
```diff
@@ -22,50 +24,98 @@ const styles = StyleSheet.create({
...
+                </NavLink>{" "}
+              </>
+            ))}
           <br />
           <b className={css(styles.style_margin)}>Telemetry address name</b>
-          {telemetryAddressName.type}
-          {telemetryAddressName.plan ? ", " + telemetryAddressName.plan : ""}
+          {telemetryAddresses &&
+            telemetryAddresses.length > 0 &&
+            telemetryAddresses.map((address: string) => (
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/project-detail/components/GeneralInfo/GeneralInfo.tsx:89]: can we try to create generic component/function in same file  for NavLink  telemetryAddresses and commandAddresses. both have same kind of logic. Component/function name can be like  NavLinkList.

Comment (drcoolsanjeev) [reply console/console-init/ui/src/modules/project-detail/components/GeneralInfo/GeneralInfo.tsx:89]: Will create in future when we declare the routes.

**Pull request merged**